### ISSUE TITLE
Absent Error Message when following Introduction

### ISSUE THREAD

**Issue by cbemeijer** [OpenModelica/OpenModelica#10324]
### Description
When following the [introduction on the Open Modelica site](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/introduction.html#interactive-session-with-examples) in one of the first examples that is meant to illustrate error messages "Example Session 1" renders no error message.

### Steps to reproduce

In the OMShell ():

```
>>> model A Integer t = 1.5; end A; 
{A}

>>> instantiateModel(A)
""

>>> getErrorString()
""
```

### Expected behaviour
According to [Example Session 1](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/introduction.html#example-session-1) this is expected:
```
>>> model A Integer t = 1.5; end A;  //The type is Integer but 1.5 is of Real Type
{A}
>>> instantiateModel(A)
""
"[<interactive>:1:9-1:23:writable] Error: Type mismatch in binding t = 1.5, expected subtype of Integer, got type Real.
"
```

### Version and OS
Run in OMShell: OpenModelica v1.21.0-dev-301-gef60d40eff (64-bit)
OS: Windows 11 home, version 22H2.

### Additional context
I write this bug report after i had [this exchange](https://github.com/OpenModelica/OpenModelica/discussions/10311) in the discussion section. One could argue that the problem is with the introduction instead of the shell but I am partial to the view that the problem is with the shell.

**Comment by perost** [comment]
> Maybe a better title would have been "Absent Error Message When assigning a Real value to an Integer Variable."

Not really, the issue seems to be entirely with OMShell somehow. If you try the example in any other tool, or even do the same thing in a .mos-script, you will get the appropriate error message. So it's not that we don't give an error message for this case, it's that `getErrorString()` doesn't seem to work properly in OMShell for some reason.

**Comment by perost** [comment]
The issue should hopefully be fixed by #10326. The issue was that the expression evaluator called `getErrorString` and stored the error, but this was then never used and instead OMShell tried to use `getErrorString` again to get the error. But since the first `getErrorString` cleared the error buffer this would always return nothing.

I'm not entirely sure what the intended behaviour was here, it seems to have been broken for at least the past six years. But I changed it to behave like the examples in the introduction, i.e. any error message is immediately shown instead of requiring the user to call `getErrorString` themselves.

**Comment by cbemeijer** [comment]
It works! thanks!
